.github/workflows/test.yml

```diff
@@ -0,0 +1,43 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - master
+      # dependabot/** branches are generated by https://github.com/dependabot
+      - dependabot/**
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  # separate job to set as required in branch protection,
+  # as the build names above change each time Node versions change
+  test:
+    name: "Test @ GO ^1.18"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.18"
+          check-latest: true
+      - name: Super Linter Code Base
```

Comment:
If I am not totally wrong, you missed separating the linting from the tests.
Please review.

go matrix:
1.17
1.18

linting under 1.18

Reply:
Okay got it, was a bit confused about linting, thanks for clarification.